src/Tests/Feature/Web/ThreadDeleteTest.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace TeamTeaTime\Forum\Tests\Feature\Web;
+
+use Illuminate\Foundation\Auth\User;
+use Orchestra\Testbench\Factories\UserFactory;
+use TeamTeaTime\Forum\Database\Factories\PostFactory;
+use TeamTeaTime\Forum\Database\Factories\ThreadFactory;
+use TeamTeaTime\Forum\Models\Category;
+use TeamTeaTime\Forum\Models\Thread;
+use TeamTeaTime\Forum\Models\Post;
+use TeamTeaTime\Forum\Support\Web\Forum;
+use TeamTeaTime\Forum\Tests\FeatureTestCase;
+
+class ThreadDeleteTest extends FeatureTestCase
+{
+    private string $route = 'thread.delete';
+
+    private PostFactory $postFactory;
+    private ThreadFactory $threadFactory;
+
+    private User $user;
+    private Thread $thread;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+        
+        $this->threadFactory = ThreadFactory::new();
+        $this->postFactory = PostFactory::new();
+
+        $this->user = UserFactory::new()->createOne();
+        $this->thread = $this->threadFactory->createOne(['author_id' => $this->user->getKey()]);
+    }
+
+    /** @test */
+    public function should_302_when_not_logged_in()
+    {
+        $response = $this->delete(Forum::route($this->route, $this->thread), []);
+        $response->assertStatus(302);
+    }
+
+    /** @test */
+    public function should_404_with_invalid_thread_id()
+    {
+        $thread = $this->thread;
+        $thread->id++;
+        $response = $this->actingAs($this->user)
+            ->delete(Forum::route($this->route, $thread), []);
+        $response->assertStatus(404);
+    }
+
+    /** @test */
+    public function should_soft_delete_by_default()
+    {
+        $thread = $this->threadFactory->createOne(['author_id' => $this->user->getKey()]);
+        $this->postFactory->createOne(['thread_id' => $thread->getKey(), 'author_id' => $this->user->getKey()]);
+
+        $this->actingAs($this->user)->delete(Forum::route($this->route, $thread), []);
+
+        // Accounting for $this->thread
+        $this->assertEquals(1, Thread::count());
+        $this->assertEquals(0, Post::count());
+        $this->assertEquals(2, Thread::withTrashed()->count());
+        $this->assertEquals(1, Post::withTrashed()->count());
+    }
+
+    /** @test */
+    public function should_delete_all_posts_inside_the_thread()
+    {
+        $numPosts = 2;
+
+        $thread = $this->threadFactory->createOne(['author_id' => $this->user->getKey()]);
+        $this->postFactory->count($numPosts)->create(['thread_id' => $thread->getKey(), 'author_id' => $this->user->getKey()]);
+
+        $this->assertEquals($numPosts, Post::count());
```

Comment:
Don't need to test this - you should already be tested elsewhere that post count is successfully updated.